src/App.jsx

```diff
@@ -1,104 +1,117 @@
-    switch(true){
-      case (nextQuestionId === 'init'):
-          setTimeout(() => this.displayNextQuestion(nextQuestionId), 500);
-          break; 
+    switch (true) {
+      case nextQuestionId === "init":
+        setTimeout(() => this.displayNextQuestion(nextQuestionId), 500);
+        break;
 
-      case (nextQuestionId === 'contact'):
+      case nextQuestionId === "contact":
         this.handleClickOpen();
         break;
 
-
       default:
         const chats = this.state.chats;
-        chats.push({ 
+        // answersに一旦これまでの回答を格納
+        const answers = this.state.selectedAnswers;
+        this.setState({
+          // answersに今回選択された解答を追加
+          selectedAnswers: [...answers, selectedAnswer],
+        });
+        chats.push({
           text: selectedAnswer,
-          type: 'answer'
-        })
-        this.setState( {
-          chats: chats
-        })
-        setTimeout(  () => this.displayNextQuestion(nextQuestionId), 1000);
+          type: "answer",
+        });
+        this.setState({
+          chats: chats,
+        });
+        setTimeout(() => this.displayNextQuestion(nextQuestionId), 1000);
         break;
     }
-  }
+  };
 
-   handleClickOpen = () => {
-    this.setState( {open: true });
+  handleClickOpen = () => {
+    this.setState({ open: true });
   };
-   handleClose = () => {
-    this.setState( {open: false });
+  handleClose = () => {
+    this.setState({ open: false });
   };
 
   componentDidMount() {
-      const initAnswer = ""
-      this.selectAnswer(initAnswer, this.state.currentId)
-     }
-
-     componentDidUpdate(prevProps, prevState, snapshot, ) {
-     const scrollArea = document.getElementById('scroll-area')
-        if (scrollArea) {
-          scrollArea.scrollTop = scrollArea.scrollHeight
-        }
-      }
+    const initAnswer = "";
+    this.selectAnswer(initAnswer, this.state.currentId);
+  }
 
-     render () {
-         return (
-           <section className="c-sention">
-             <div className="c-box">
-               <Chats chats={this.state.chats}/>
-               <AnswersList answers={this.state.answers} select={this.selectAnswer} />
-               <FormDialog open={this.state.open} handleClose={this.handleClose} />
-             </div>
-           </section> 
-   
-           );
-       }
+  componentDidUpdate(prevProps, prevState, snapshot) {
+    const scrollArea = document.getElementById("scroll-area");
+    if (scrollArea) {
+      scrollArea.scrollTop = scrollArea.scrollHeight;
     }
-   
+  }
+
+  render() {
+    return (
+      <section className="c-sention">
+        <div className="c-box">
+          <Chats chats={this.state.chats} />
+          <AnswersList
+            answers={this.state.answers}
+            select={this.selectAnswer}
+          />
+          <FormDialog
+            open={this.state.open}
+            handleClose={this.handleClose}
+            selectedAnswers={this.state.selectedAnswers}
```

Comment:
ここでモーダルのコンポーネントに選択された回答を渡してる。

Reply:
101行目　{this.state.selectedAnswers}
selectedAnswers={selectedAnswers}でも解答を渡せそうな気がするのですが

この場合のthis.stateとはどんな効果があるんでしょうか？
自分は、今回選択された解答を更新するために一旦、初期化して新しい値を更新してるのかと思います。

参照記事
https://qiita.com/rio_threehouse/items/7632f5a593cf218b9504

Reply:
```suggestion
            selectedAnswers={selectedAnswers}
```
selectedAnswersの場合でもモーダルに反映できた感じかな？

Reply:
selectedAnswers={selectedAnswers}にすると下記エラーにあたりました。

```
ERROR in [eslint] 
src/App.jsx
  Line 101:30:  'selectedAnswers' is not defined  no-undef
Search for the keywords to learn more about each error.
webpack compiled with 1 error and 1 warning
```

eslintが影響してるとのエラーですが、App.jsxの10~18行目
```
this.state = {
      answers: [],
      chats: [],
      // 選択した解答を配列に追加
      selectedAnswers: [],
      currentId: "init",
      dataset: defaultDataset,
      open: false,
    };
```
で定義？してるのにthis.stateを削除することで、selectedAnswersの配列を呼び出せなないがためのエラーだと思います。

Reply:
そうやね！
>selectedAnswers={selectedAnswers}でも解答を渡せそうな気がするのですが

↑はできないことがわかったね。
今定義しているのはthis.stateの中にselectedAnswersを定義しているわけであって
this.state.selectedAnswersじゃないと呼び出せない。

>この場合のthis.stateとはどんな効果があるんでしょうか？
自分は、今回選択された解答を更新するために一旦、初期化して新しい値を更新してるのかと思います。

https://qiita.com/mzmz__02/items/e6d7a6e84edc31ae2c9f
この記事とか見てみるといいかも。クラスコンポーネントの書き方やから
深く理解する必要はないけど、簡単にいうとthis.state内に
状態管理したい（ユーザーの操作によって値が変化する）ものを格納するルールがあるからかな。